### ISSUE TITLE
Load modules

### ISSUE THREAD

**Issue by thekid** [xp-framework/rfc#306]
# Scope of Change

This RFC suggests adding a command line switch to load modules: `xp -m [path-to-module]` being the (rough) equivalent of `xp -cp [path-to-module]/src/main/php`.
# Rationale

A central idea of XP 7 is framework modularity. This means many packages previously part of the framework now need to be referenced by either adding them to the class path or by spelling out all the class path entries (plus those of transitive dependencies) on the command line. Running a script to query the database will require either creating a composer.json and running composer or using the longish `xp -cp ~/xp/rdbms/src/main/php -cp ~/xp/networking/src/main/php -cp ~/xp/logging/src/main/php  -cp ~/xp/math/src/main/php Script`.
# Functionality

Syntax: `xp [-m module [-m module [-m ...]]] <options> <Class>`

The `module` argument references a module by its top-level directory. The determined directory is then scanned for path files which finally make up the class path.
# Security considerations

n/a
# Speed impact

Supplying `-m module` is slightly slower than `-cp module/src/main/php` since the first parses path files to determine the directories. As `-cp` is retained, no harm is done.
# Dependencies

Since all module paths are simply added to the `use` path, this change is compatible with XP5, XP6 and the upcoming XP7!
# Related documents
- Further shrink core: #296 - the general idea
- Path to XP: #298 - progress towards XP 7
- Extract unittests from core: #293 - one example
- Install XP Modules: #270 - this was the original idea before moving to composer

**Comment by thekid** [comment]
_The following was removed because [I wasn't too happy](https://github.com/xp-framework/xp-runners/pull/35#issuecomment-168181933) with the complexity it added:_

---

The `module` argument references a module by its qualified name `[vendor]/[package]`. If module is an existent directory, all path files inside it are scanned for class path entries. Otherwise, the `module` path defined via `xp.ini` is scanned. It defaults to [$COMPOSER_HOME](https://getcomposer.org/doc/03-cli.md#composer-home)/vendor/{vendor}/{name}.

### Composer integration

The module path defaults to [$COMPOSER_HOME](https://getcomposer.org/doc/03-cli.md#composer-home)/vendor/{vendor}/{name}, which is used by `composer global require ...`:
- **Un*x**: `/home/$USER/.composer/vendor/{vendor}/{name}`
- **Mac**: `/Users/$USER/.composer/vendor/{vendor}/{name}`
- **Windows**: `C:\Users\$USER\AppData\Roaming\Composer\vendor\{vendor}\{name}`
- **Cygwin**: First checks for Un*x path, then Windows

### Configuring the module path

The path can also be set inside xp.ini as follows:

``` ini
[runtime]
date.timezone=Europe/Berlin
modules=~/devel/{vendor}/{name}
extension=php_com_dotnet.dll
```

The module `xp-framework/logging` will assumed to be in ~/devel/xp-framework/logging, which could contain a git checkout.

---

_We're able to create the new relatively simple approach first without breaking anything in a way where this could not be added later._

**Comment by thekid** [comment]
:green_heart: There is no impact on the framework from this RFC and no change necessary, removed "framework" label and added note that this is compatible with XP5, XP6 and the upcoming XP7.

**Comment by thekid** [comment]
:shipit: **Released in XP Runners [6.0.0](https://github.com/xp-framework/xp-runners/releases/tag/v6.0.0)**
